src/TimeSeries/widget/TimeSeries.ts

```diff
@@ -0,0 +1,191 @@
+            widthUnit: this.widthUnit,
+            xAxisFormat: this.xAxisFormat,
+            xAxisLabel: this.xAxisLabel,
+            yAxisFormat: this.yAxisFormat,
+            yAxisLabel: this.yAxisLabel
+        };
+    }
+
+    postCreate() {
+        this.dataStore = {};
+        this.dataStore.series = this.seriesConfig.reduce((previousValue: any, currentValue: SeriesConfig ) => {
+            return previousValue[currentValue.seriesKey] = [];
+        }, {});
+
+        this.updateRendering();
+    }
+
+    update(object: mendix.lib.MxObject, callback?: Function) {
+        this.contextObject = object;
+        if (this.contextObject && this.checkConfig()) {
+            this.updateData(() => {
+                this.dataLoaded = true;
+                this.updateRendering(callback);
+            });
+        } else {
+            this.updateRendering(callback);
+        }
+        this.resetSubscriptions();
+    }
+
+    uninitialize(): boolean {
+        unmountComponentAtNode(this.domNode);
+        return true;
+    }
+
+    private updateData(callback: Function) {
+        const series = this.seriesConfig[0];
+        // TODO: do this in a async parallel way for all series, in the future.
+        if (series.seriesSource === "xpath" && series.seriesEntity) {
+            this.fetchDataFromXpath(series, (data: mendix.lib.MxObject[]) => {
+                this.setDataFromObjects(data, series);
+                callback();
+            });
+        } else if (series.seriesSource === "microflow" && series.dataSourceMicroflow) {
+             this.fetchDataFromMicroflow(series, (data: mendix.lib.MxObject[]) => {
+                 this.setDataFromObjects(data, series);
+                 callback();
+             });
+        } else {
+            logger.error(this.id + ".updateData unknown source or error in widget configuration");
+            callback();
+        }
+    }
+
+    /**
+     * Validate the widget configurations from the modeler
+     */
+    private checkConfig() {
+        let valid = true;
+        const incorrectSeries = this.seriesConfig.filter(series =>
+            (series.seriesSource === "microflow" && !series.dataSourceMicroflow)
+        );
+
+        if (incorrectSeries.length) {
+            valid = false;
+            mx.ui.error("Configuration error for series : '" + incorrectSeries[0].seriesKey +
+                "'. Source is set to 'Microflow' but 'Source - microflow' is missing ", true);
+        }
+
+        return valid;
+    }
+
+    private updateRendering (callback?: Function) {
+        render(createElement(TimeSeries, this.createProps()), this.domNode);
+
+        mxLang.nullExec(callback);
+    }
+
+    private resetSubscriptions () {
+        mx.data.unsubscribe(this.handle);
+        this.handle = 0;
+
+        if (this.contextObject) {
+            this.handle = this.subscribe({
+                callback: (guid: string) => {
+                    this.updateRendering();
```

Comment:
Why is this necessary? We don't use anything from the contextObject which would have any impact  on the time series.